File: Task3/Task3.cs
using System;
using System.Collections.Generic;

class Program {
    static (int sum, int product) SumAndProductOfDigits(int n) {
        int sum = 0, product = 1;
        while (n > 0) {
            int digit = n % 10;
            sum += digit;
            product *= digit;
            n /= 10;
        }
        return (sum, product);
    }

    static void Main() {
        Console.Write("Кол-во чисел: ");
        int n = int.Parse(Console.ReadLine());
        List<int> indices = new List<int>();

        for (int i = 0; i < n; i++) {
            Console.Write($"{i} число: ");
            int num = int.Parse(Console.ReadLine());
            var (sum, product) = SumAndProductOfDigits(num);
            if (sum < product) {
                indices.Add(i);
            }
        }
        Console.WriteLine(string.Join(" ", indices));
    }
}

File: Task3/Task3.cpp
#include <iostream>
#include <vector>

using namespace std;

void sumAndProductOfDigits(int n, int& sum, int& product) {
    sum = 0;
    product = 1;
    while (n > 0) {
        int digit = n % 10;
        sum += digit;
        product *= digit;
        n /= 10;
    }
}

int main() {
    int n;
    int num;
    cout << "Кол-во чисел: ";
    cin >> n;
    vector<int> numbers;
    for (int i = 0; i < n; i++) {
        cout << i << " число: ";
        cin >> num;
        numbers.push_back(num);
    }
    
    for (size_t i = 0; i < numbers.size(); ++i) {
        int sum, product;
        sumAndProductOfDigits(numbers[i], sum, product);
        if (sum < product) {
            cout << i << " ";
        }
    }
    cout << endl;
    return 0;
}